README.md

```diff
@@ -41,4 +41,10 @@ if err != nil {
 
 log.Println(string(v))
 // => Practice
+
+// Iterate over the database
+db.Each(func(key, value []byte) error {
```

Comment:
Instead of this, can we add do something like the "stateful iterators" (the fourth option) described [here](http://ewencp.org/blog/golang-iterators/)?

That's the most similar to the stdlib (see [scanner](https://golang.org/pkg/bufio/#Scanner), [sql](https://golang.org/pkg/database/sql/#Rows), etc), and I like it a bit better.

Then the example would look like:

```go
iter := db.Iter()
for iter.Next() {
   log.Printf("The key %s has a value of length %d\n", iter.Key(), iter.Value())
}

if iter.Err() != nil {
  log.Fatal(err)
}
```